lib/engine/game/g_1829/step/buy_sell_par_shares_companies.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/buy_sell_par_shares_companies'
+
+module Engine
+  module Game
+    module G1829
+      module Step
+        class BuySellParSharesCompanies < Engine::Step::BuySellParSharesCompanies
+          def actions(entity)
+            return [] unless entity == current_entity
+            return ['sell_shares'] if must_sell?(entity)
+
+            actions = []
+            actions << 'buy_shares' if can_buy_any?(entity)
+            actions << 'buy_company' if can_buy_any_companies?(entity)
+            actions << 'sell_shares' if can_sell_any?(entity)
+
+            actions << 'pass' unless actions.empty?
+            actions
+          end
+
+          def can_buy_any_companies?(entity)
+            return false if
+              @game.num_certs(entity) >= @game.cert_limit
+
+            @game.companies.any? { |c| c.owner == @bank && !c.closed? }
```

Comment:
duplicate

Reply:
sorry,  I'm not so good in ruby. the first definition is in step/draft.rb. How can i access this definition from buy_sell_par_shares_companies.rb? Can you help me?

Reply:
you need to put the definition in game and then call it